components/tools/OmeroPy/src/omero/plugins/metadata.py

```diff
@@ -554,6 +565,87 @@ def populateroi(self, args):
                 meas = ctx.get_measurement_ctx(i)
                 meas.parse_and_populate()
 
+    def pixelsize(self, args):
+        "Set physical pixel size"
+        md = self._load(args)
+        client, conn = self._clientconn(args)
+
+        units = {
+            'nm': UnitsLength.NANOMETER,
+            'um': UnitsLength.MICROMETER,
+            'mm': UnitsLength.MILLIMETER,
+            'cm': UnitsLength.CENTIMETER,
+            'm': UnitsLength.METER,
+            'km': UnitsLength.KILOMETER
+        }
+
+        if args.unit not in units:
+            self.ctx.die(100, "%s not recognized as valid unit." % args.unit)
+
+        if not args.x and not args.y and not args.z:
+            self.ctx.die(100, "No pixel sizes specified.")
+
+        if md.get_type() == "Screen":
+            q = """SELECT pix FROM Pixels pix, WellSample ws, Plate p,
+                   ScreenPlateLink spl WHERE
+                   spl.child=p AND pix.image=ws.image AND ws.well.plate=p AND
+                   spl.parent.id=:id"""
+        elif md.get_type() == "Plate":
+            q = """SELECT pix FROM Pixels pix, WellSample ws WHERE
+                   pix.image=ws.image AND ws.well.plate.id=:id"""
+        elif md.get_type() == "PlateAcquisition":
+            q = """SELECT pix FROM Pixels pix, WellSample ws WHERE
+                   pix.image=ws.image AND ws.plateAcquisition.id=:id"""
+        elif md.get_type() == "Well":
+            q = """SELECT pix FROM Pixels pix, WellSample ws WHERE
+                   pix.image=ws.image AND ws.well.id=:id"""
+        elif md.get_type() == "Project":
+            q = """SELECT pix FROM Pixels pix, DatasetImageLink dil,
+                   ProjectDatasetLink pdl WHERE dil.child=pix.image AND
+                   dil.parent=pdl.child AND pdl.parent.id=:id"""
+        elif md.get_type() == "Dataset":
+            q = """SELECT pix FROM Pixels pix, DatasetImageLink dil WHERE
+                   dil.child=pix.image AND dil.parent.id=:id"""
+        elif md.get_type() == "Image":
+            q = """SELECT pix FROM Pixels pix WHERE pix.image.id=:id"""
+        else:
+            raise Exception("Not implemented for type %s" % md.get_type())
```

Comment:
This is fine but just to mention that another option is http://downloads.openmicroscopy.org/latest/omero5.4/api/slice2html/omero/cmd/FindChildren.html#FindChildren to find their IDs with `typesOfChildren = ['Pixels']` and `stopBefore = ['Roi']`.

Reply:
Do we have an example somewhere which uses `FindChildren`? Can't get my head around how to use that.

Reply:
```python
from omero.gateway import BlitzGateway
from omero.cmd import FindChildren
from omero.callbacks import CmdCallbackI

conn = BlitzGateway(stuff)  # I suggest user-2 on eel
conn.connect()

req = FindChildren(targetObjects={'Project': [1]},
                   typesOfChildren=['Pixels'], stopBefore=['Roi'])
prx = conn.c.sf.submit(req)
print CmdCallbackI(conn.c, prx).loop(500, 500)

conn._closeSession()
```
and see many IDs of Pixels that are in Project 1.

Reply:
Thanks! That works. But I think I leave the the hql query for now, because it loads the Pixel objects directly. Otherwise I'd have to get the ids first with `FindChildren` and the load the Pixel objects from the id list.

Reply:
Btw, that example would be nice to have on https://docs.openmicroscopy.org/omero/5.4.4/developers/Python.html :-)

Reply:
Leaving someone else to record or migrate the Pythong example.

Reply:
https://trello.com/c/8oJXeUPY/625-findparents-findchildren-example